feat: UX: Multichain: Release the send page

darkwing opened the pull request:
## **Description**

Removes the feature flag surrounding the new send flow, as well as removes legacy send files.

## **Related issues**

N/A

## **Manual testing steps**

1. Go to Send page
2. See the new UI
3. Complete a send
4. Go to an NFT page
5. Click send
6. Send the NFT appropriately

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

darkwing committed:
UX: Multichain: Release the send page

Restore domain-input

Fix test for send error

Fix max button test

E2Es: Update amount selector, Next to Continue button

Remove header for send flow

Fix address book and ENS tests

Update snap-account-transfers.spec.ts and helpers.js

Update change-language.spec.js

Fix custom-token-send-transfer.spec.js

Fix missing NFT icon

Make send-nft.spec.js pass, set send volume to 1 when NFT

Fix send-edit.spec.js

Fix address-book.spec.js

Release the address book changes

darkwing commented:
Blocked on: https://consensyssoftware.atlassian.net/browse/MMASSETS-131

darkwing commented:
Presently blocked by: https://github.com/MetaMask/metamask-extension/pull/22551

darkwing committed:
Restore localizations

darkwing committed:
Fixes #22986 - UX - Multichain - Send flow - Show address in account … (#23025)

The shortened address should display in the AccountPicker within the new
Send flow

Fixes: https://github.com/MetaMask/metamask-extension/issues/22986

1. Open the new send flow
2. See the account address under the account name

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

N/A

<img width="420" alt="SCR-20240216-mxwm"
src="https://github.com/MetaMask/metamask-extension/assets/46655/ca6dbe2b-c136-474a-860d-970ccdad9f86">

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it
is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
- [ ] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

darkwing committed:
Fix #22938 - UX - Multichain - Send Flow - Update ENS address upon network change (#23024)

## **Description**

Changing the network mid-send can effect ENS domains in a negative way.
Namely, an ENS domain can have a different addresses per chain. This PR
ensure that ENS lookups are done when the user changes chains.

## **Related issues**

Fixes:  https://github.com/MetaMask/metamask-extension/issues/22938

## **Manual testing steps**

### Scenario 1
0.  Have an ENS with two different address for Mainnet and Goerli
1.  Enter the new send flow
2. Type in the ENS on Mainnet -- see a search result with the correct
address
3. Change the network picker to Goerli
4. See the address change in the search result

### Scenario 2
0.  Have an ENS with two different address for Mainnet and Goerli
1.  Enter the new send flow
2. Type in the ENS on Mainnet -- see a search result with the correct
address
3. Click the search result (you'll go to the Amount screen)
4. Change the network picker to Goerli
5. See the address change in the recipient field change
6. Execute the transaction
7. Ensure token sent to correct address

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

(In the GitHub issue)

### **After**

Coming....

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it
is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
- [ ] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

hesterbruikman commented:
@darkwing Does it make sense to you if I add a `DO NOT MERGE` label and add the Send flow bug fixes epic [#2106](https://github.com/MetaMask/MetaMask-planning/issues/2106) as a blocker?

darkwing committed:
Fix test/e2e/tests/send-eth.spec.js

darkwing committed:
Fix send-nft.sepc.js

darkwing committed:
Fix lint in routes test

darkwing committed:
Rebased test helper fix

darkwing committed:
Fix send-hex-address.spec.js

darkwing committed:
Fix custom-token-send-transfer.spec.js

darkwing committed:
Fix #23304 - Only use Confusable during send with ENS

darkwing committed:
Fix toChecksumHexAddress import

darkwing committed:
Fix gas tests

darkwing committed:
Remove more feature flags from test

darkwing committed:
Fix lint

darkwing committed:
Add back data-id for asset picker error

darkwing committed:
Revert "Fix #22938 - UX - Multichain - Send Flow - Update ENS address upon network change (#23024)"

This reverts commit f17a26b9810150f6bccb665e6eb4774a8eda4a3e.

darkwing committed:
Remove unwanted, incomplete test

darkwing committed:
Restore @micaelae's useTokenTracker fix

cryptotavares commented on app/_locales/pl/messages.json:
```diff
@@ -127,6 +127,9 @@
     "message": "Anuluj"
   },
+  "cancelEdit": {
+    "message": "Cancelar edição"
```
This is a portuguese translation... pretty sure it shouldn't be on the polish translations file

cryptotavares commented on test/e2e/helpers.js:
```diff
@@ -760,36 +760,9 @@ const sendScreenToConfirmScreen = async (
   await driver.fill('.unit-input__input', quantity);
-  if (process.env.MULTICHAIN) {
-    // check if element exists and click it
-    await driver
-      .findElement({
-        text: 'I understand',
-        tag: 'button',
-      })
-      .then(
-        (_found) => {
-          driver.clickElement({
-            text: 'I understand',
-            tag: 'button',
-          });
-        },
-        (error) => {
-          console.error('Element not found.', error);
-        },
-      );
```
I don't think that this should be removed. The toolTipApproveButton label is used in the [SendPageContent component](https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/multichain/pages/send/components/content.tsx#L83-L103), that is used on the new send page.

cryptotavares commented on test/e2e/helpers.js:
```diff
@@ -760,36 +760,9 @@ const sendScreenToConfirmScreen = async (
-    });
-  }
+  await driver.clickElement({ text: 'Continue', tag: 'button' });
+  await driver.clickElement({ text: 'Confirm', tag: 'button' });
```
Remove clicking on the confirm button. We do not want to submit the transaction. The `sendScreenToConfirmScreen` helper is used for ppom tests, and on those tests we use mainnet (for ppom support.. we mock all network requests)

cryptotavares commented on ui/pages/routes/routes.component.js:
```diff
@@ -371,11 +370,7 @@ export default class Routes extends Component {
-          component={process.env.MULTICHAIN ? SendPage : SendTransactionScreen}
-          exact
-        />
+        <Authenticated path={SEND_ROUTE} component={SendPage} exact />
```
With this it seems that we are no longer using the SendTransactionScreen page anywhere. Are we going to remove all the dead code that will linger around after this PR?

digiwand reviewed (commented):
code lgtm! +1 on @cryptotavares's comments

darkwing committed:
Remove confirmation of sendScreenToConfirmScreen to avoid PPOM issues

darkwing committed:
Revert localization changes

darkwing committed:
UX: Multichain: Send Flow: Disable account picker for edits (#23708)

## **Description**

Disables the account picker in the new send flow during an Edit

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/23708?quickstart=1)

## **Related issues**

Fixes: N/A

## **Manual testing steps**

1. `yarn start`
2. Start a send
3. Continue
4. Click Edit to edit the send
5. See the account picker disabled

## **Screenshots/Recordings**

### **Before**

N/A

### **After**

<img width="573" alt="SCR-20240325-mnda"
src="https://github.com/MetaMask/metamask-extension/assets/46655/ad185a0c-8fc3-4174-95de-61731f1163e6">


## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

darkwing committed:
Fix jest

darkwing committed:
remove flag check from test/e2e/tests/ppom/ppom-blockaid-alert-simple-send.spec.js

bowensanders reviewed (dismissed):
[Snaps] LGTM :) ✅

bowensanders reviewed (approved):
LGTM
